axelrod/strategies/adaptive.py

```diff
@@ -0,0 +1,49 @@
+from axelrod import Actions, Player, Game
+
+C, D = Actions.C, Actions.D
+
+
+class Adaptive(Player):
+    """Start with a specific sequence of C and D, then play the strategy that
+    has worked best, recalculated each turn."""
+
+    name = 'Adaptive'
+    classifier = {
+        'memory_depth': float('inf'),  # Long memory
+        'stochastic': False,
+        'makes_use_of': set(),
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self):
+        Player.__init__(self)
+        self.initial_plays = [C] * 6 + [D] * 5
+        self.reset()
+
+    def score_last_round(self, opponent):
+        try:
+            game = self.match_attributes["game"]
+        except AttributeError:
+            game = Game()
+        if len(self.history):
+            last_round = (self.history[-1], opponent.history[-1])
+            scores = game.score(last_round)
+            self.scores[last_round[0]] += scores[0]
+
+    def strategy(self, opponent):
+        # Update scores from the last play
+        self.score_last_round(opponent)
+        # Begin by playing the sequence C,C,C,C,C,C,D,D,D,D,D
```

Comment:
Using the initial sequence decorator would not make changing the initial sequence available to the user (and note that for the handshake strategy it wouldn't do the trick as we need to know the length of the handshake so to speak).

That's my request (for here and handshake): make these nice generic strategies, the default value of which is one that adds to the completeness of the library. 

I still don't understand why we wouldn't do this, even if it's not used it's nice to have and seems like good practice (a part from adding one small extra test it's not that much extra work). **However, this should not delay this PR from going in** it's a simple thing that could be added at a later time. If there's opposition perhaps easiest to just leave it as it is :)